applications/GeoMechanicsApplication/custom_utilities/builder_and_solver_factory.hpp

```diff
@@ -0,0 +1,43 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+#pragma once
+
+#include <memory>
+#include "solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h"
+#include "solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver.h"
+
+namespace Kratos
+{
+
+template<class TSparseSpace, class TDenseSpace, class TLinearSolver>
+class BuilderAndSolverFactory
+{
+
+public:
+    static std::shared_ptr<BuilderAndSolver<TSparseSpace, TDenseSpace, TLinearSolver>> Create(const Parameters& rSolverSettings,
```

Comment:
Just like you did for the `ConvergenceCriteriaFactory`, you could define an alias that simplifies the return type of `Create`.
```suggestion
    using BuilderAndSolverType = BuilderAndSolver<TSparseSpace, TDenseSpace, TLinearSolver>;
    static std::shared_ptr<BuilderAndSolverType> Create(const Parameters& rSolverSettings,
```

Reply:
Agreed, done